src/api.rs

```diff
@@ -71,7 +76,63 @@ pub trait Introspectable {
         unimplemented!();
     }
 
+    // toggle an event interception
+    fn toggle_intercept(
+        &mut self,
+        _vcpu: u16,
+        _intercept_type: InterceptType,
+        _enabled: bool,
+    ) -> Result<(), Box<dyn Error>> {
+        unimplemented!();
+    }
+
+    // listen and return the next event, or return None
+    fn listen(&mut self, _timeout: u32) -> Result<Option<Event>, Box<dyn Error>> {
+        unimplemented!();
+    }
+
+    fn reply_event(
+        &mut self,
+        _event: &Event,
```

Comment:
Just a thought: It would make sense to consume the event here, right? This would automatically make sure that the user replies only once per event. Now we just need to ensure that the user sends a reply at all. ;)